src/main/kotlin/io/snyk/plugin/snykcode/core/SnykCodeUtils.kt

```diff
@@ -27,35 +24,32 @@ class SnykCodeUtils private constructor() : DeepCodeUtilsBase(
         scLogger.logInfo("allProjectFiles requested")
         val project = PDU.toProject(projectO)
         val progressIndicator = ProgressManager.getInstance().progressIndicator ?: null
-        val allVirtualFiles = mutableSetOf<VirtualFile>()
+        val files = mutableSetOf<SnykCodeFile>()
 
         val cancelled = !ProjectRootManager.getInstance(project).fileIndex.iterateContent {
             if (!it.isDirectory) {
-                allVirtualFiles.add(it)
+                files.add(SnykCodeFile(project, it))
             }
             return@iterateContent progressIndicator?.isCanceled != true
         }
         if (cancelled) {
-            allVirtualFiles.clear()
+            files.clear()
         }
-        scLogger.logInfo("allProjectFiles scan finished. Found ${allVirtualFiles.size} files")
-
-        val psiManager = PsiManager.getInstance(project)
+        scLogger.logInfo("allProjectFiles scan finished. Found ${files.size} files")
 
         return RunUtils.computeInReadActionInSmartMode(
             project,
-            Computable { allVirtualFiles.filter { it.isValid }.mapNotNull(psiManager::findFile) }
-        ) ?: emptyList<PsiFile>()
+            Computable { files.filter { it.virtualFile.isValid } }
+        ) ?: emptyList<SnykCodeFile>()
     }
 
-    override fun getFileLength(file: Any): Long = PDU.toPsiFile(file).virtualFile.length
+    override fun getFileLength(file: Any): Long = PDU.toVirtualFile(file).length
 
-    override fun getFileExtention(file: Any): String = PDU.toPsiFile(file).virtualFile.extension ?: ""
+    override fun getFileExtention(file: Any): String = PDU.toVirtualFile(file).extension ?: ""
 
     override fun isGitIgnoredExternalCheck(file: Any): Boolean {
-        val psiFile = PDU.toPsiFile(file)
-        val virtualFile = psiFile.virtualFile ?: return false
-        return ChangeListManager.getInstance(psiFile.project).isIgnoredFile(virtualFile)
+        require(file is SnykCodeFile) { "File $file must be of type SnykCodeFile but is ${file.javaClass.name}" }
+        return ChangeListManager.getInstance(PDU.instance.getProject(file) as Project).isIgnoredFile(file.virtualFile)
```

Comment:
`toSnykCodeFile` and `toProject` usage here to avoid (log at least) cast exceptions